sdk/storage/src/queue/requests/clear_messages_builder.rs

```diff
@@ -28,7 +28,10 @@ impl<'a> ClearMessagesBuilder<'a> {
     pub async fn execute(
         &self,
     ) -> Result<ClearMessagesResponse, Box<dyn std::error::Error + Sync + Send>> {
-        let mut url = self.queue_client.queue_url()?.join("messages")?;
+        let mut url = self.queue_client.queue_url()?.to_owned();
+        url.path_segments_mut()
+            .map_err(|_| "Invalid queue URL")?
+            .push("messages");
```

Comment:
This code is being repeated a lot. Is there anyway to share this in all the places it's being used?

Reply:
Maybe, but if you have a close look, you'll see that sometimes I did not want to change the returned error type and used `.map_err(|_| url::ParseError::SetHostOnCannotBeABaseUrl)`. This is not fully correct, but it is the closest variant I found to raise an error which should never happen as the URL was parsed already.

If you want to have a bit of harmony, what would be your preffered implementation?

Reply:
Meanwhile, I pushed the beginning of the refactoring, tell me what you think of it.